src/vm/finalizerthread.cpp

```diff
@@ -1207,8 +1207,13 @@ BOOL FinalizerThread::FinalizerThreadWatchDog()
 #endif //BACKGROUND_GC
 
         _ASSERTE ((g_fEEShutDown & ShutDown_Finalize1) || g_fFastExitProcess);
+
+#ifndef FEATURE_CORECLR
         ThreadSuspend::SuspendEE(ThreadSuspend::SUSPEND_FOR_SHUTDOWN);
 
+        // FinalizerThreadWatchDogHelper does not time out waiting for the finalizer thread in CoreCLR. If another thread blocks
```

Comment:
nit pick - may be nice to move the comment in from of `FEATURE_CORECLR` ifdef to make it easier to see that it applies to CoreCLR.